thunder/core/proxies.py

```diff
@@ -1277,6 +1294,31 @@ def distparallel_type(self):
     def thunder_fsdp_padding_size(self):
         return self._thunder_fsdp_padding_size
 
+    @property
+    def version(self):
+        return self._version
+
+    @property
+    def view_of(self):
+        return self._view_of
+
+    @property
+    def is_view(self) -> bool:
+        return self.view_of is not None
```

Comment:
Why do we have to store these as properties? Seems like `version` is more reminiscent of the eager mode. Looks like all these versions and view_of can be read from a trace and packed into a datastructure of a UnionFind type...

Reply:
Yes this should be simplified. I think I drastically change the implementation following #630 and #633. I also think of using views rather than original tensors to avoid may copies based on @jjsjann123 advice